### ISSUE TITLE
pip install fails

### ISSUE THREAD

**Issue by keithbriggs** [boschmitt/tweedledum#165]
On a standard up-to-date Ubuntu 18 system:

**Comment by keithbriggs** [comment]
I think the basic problem is that setup.py does not specify that g++ 8 or higher is needed.

**Comment by boschmitt** [comment]
> I think the basic problem is that setup.py does not specify that g++ 8 or higher is needed.

Yes. It requires a compiler which supports C++17 (both core language and library features)

**Comment by keithbriggs** [comment]
So could a check for this be added to setup.py?  Otherwise you get a failure with cryptic error messages.
